The tested assertions are failing in the backport on Windows:

```
======================================================================
FAIL: test_backslash_not_separator (test.test_zipfile.TestPath)
In a zip file, backslashes are not separators.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\a\cpython\cpython\lib\test\test_zipfile.py", line 3329, in test_backslash_not_separator
    assert first.name == 'foo\\bar'
AssertionError

======================================================================
FAIL: test_unsupported_names (test.test_zipfile.TestPath)
Path segments with special characters are readable.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\a\cpython\cpython\lib\test\test_zipfile.py", line 3315, in test_unsupported_names
    assert next(contents).name == 'V: NMS.flac'
AssertionError

----------------------------------------------------------------------
```

--- a/Lib/test/test_zipfile.py
+++ b/Lib/test/test_zipfile.py
@@ -3290,6 +3290,8 @@ with zipfile.ZipFile(io.BytesIO(), "w") as zf:
         """
         data = io.BytesIO()
         zf = zipfile.ZipFile(data, "w")
+        zf.writestr("/one-slash.txt", b"content")
+        zf.writestr("//two-slash.txt", b"content")
         zf.writestr("../parent.txt", b"content")
         zf.filename = ''
         root = zipfile.Path(zf)

--- a/Lib/test/test_zipfile.py
+++ b/Lib/test/test_zipfile.py
@@ -3296,39 +3296,6 @@ with zipfile.ZipFile(io.BytesIO(), "w") as zf:
         assert list(map(str, root.iterdir())) == ['../']
         assert root.joinpath('..').joinpath('parent.txt').read_bytes() == b'content'
 
-    def test_unsupported_names(self):
-        """
-        Path segments with special characters are readable.
-
-        On some platforms or file systems, characters like
-        ``:`` and ``?`` are not allowed, but they are valid
-        in the zip file.
-        """
-        data = io.BytesIO()
-        zf = zipfile.ZipFile(data, "w")
-        zf.writestr("path?", b"content")
-        zf.writestr("V: NMS.flac", b"fLaC...")
-        zf.filename = ''
-        root = zipfile.Path(zf)
-        contents = root.iterdir()
-        assert next(contents).name == 'path?'
-        item = next(contents)
-        assert item.name == 'V: NMS.flac', item.name
-        assert root.joinpath('V: NMS.flac').read_bytes() == b"fLaC..."
-
-    def test_backslash_not_separator(self):
-        """
-        In a zip file, backslashes are not separators.
-        """
-        data = io.BytesIO()
-        zf = zipfile.ZipFile(data, "w")
-        zf.writestr(DirtyZipInfo.for_name("foo\\bar", zf), b"content")
-        zf.filename = ''
-        root = zipfile.Path(zf)
-        (first,) = root.iterdir()
-        assert not first.is_dir()
-        assert first.name == 'foo\\bar', first.name
-
 
 class DirtyZipInfo(zipfile.ZipInfo):
     """

--- a/Lib/test/test_zipfile.py
+++ b/Lib/test/test_zipfile.py
@@ -3312,7 +3312,8 @@ with zipfile.ZipFile(io.BytesIO(), "w") as zf:
         root = zipfile.Path(zf)
         contents = root.iterdir()
         assert next(contents).name == 'path?'
-        assert next(contents).name == 'V: NMS.flac'
+        item = next(contents)
+        assert item.name == 'V: NMS.flac', item.name
         assert root.joinpath('V: NMS.flac').read_bytes() == b"fLaC..."
 
     def test_backslash_not_separator(self):
@@ -3326,7 +3327,7 @@ with zipfile.ZipFile(io.BytesIO(), "w") as zf:
         root = zipfile.Path(zf)
         (first,) = root.iterdir()
         assert not first.is_dir()
-        assert first.name == 'foo\\bar'
+        assert first.name == 'foo\\bar', first.name
 
 
 class DirtyZipInfo(zipfile.ZipInfo):

